src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -228,7 +228,15 @@ module.exports = function(
   }
 
 
-  function markUpdate(workInProgress) {
+  function markUpdateIfNecessary(workInProgress) {
+    const current = workInProgress.alternate;
+    const instance = workInProgress.stateNode;
+    if (!current && typeof instance.componentDidMount !== 'function') {
+      return;
+    }
+    if (current && typeof instance.componentDidUpdate !== 'function') {
```

Comment:
Perfwise is it the same? Does the engine have to check `current` twice?

Also we should use `current === null`